Add Carbon datasource for Hadoop

gokul opened the pull request:
This component will make it possible for the user to specify Carbon datasource definitions for connecting with Hadoop-like data sources (e.g. Hadoop, HBase) using the Hadoop Configuration format.

gokul committed:
Add Hadoop Carbon data source

The Carbon data source for Hadoop can be used to read and process data source definitions used for connecting with Apache Hadoop instances.

gokul committed:
Update copyright in Hadoop data source

gokul committed:
Add Hadoop Carbon data source feature

laki88 commented on components/org.wso2.carbon.datasource.hadoop/src/main/java/org/wso2/carbon/datasource/hadoop/HadoopDataSourceUtil.java:
```diff
@@ -0,0 +1,56 @@
+            List<HadoopDataSourceProperty> properties = configuration.getProperties();
+
+            for (HadoopDataSourceProperty configEntry : properties) {
+                if (!("".equals(configEntry.getPropertyName())) && !("".equals(configEntry.getPropertyValue()))) {
```
We can use isEmpty()?

laki88 commented on components/org.wso2.carbon.datasource.hadoop/src/test/resources/testng.xml:
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
License text should be in 2017

laki88 commented on components/org.wso2.carbon.datasource.hadoop/pom.xml:
```diff
@@ -0,0 +1,133 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2016 WSO2, Inc. (http://wso2.com)
```
License text should be 2017

laki88 commented on components/org.wso2.carbon.datasource.hadoop/pom.xml:
```diff
@@ -0,0 +1,133 @@
+    </build>
+    <properties>
+        <export.package>org.wso2.carbon.datasource.hadoop.*</export.package>
+        <import.package>javax.xml.parsers;version=0.0.0,*;resolution:=optional</import.package>
```
Can't we get rid of this resolution optional?

gokul committed:
Incorporate review feedback and general fixes

- Fixed copyright year
- Switched to isEmpty()
- Added status logs on tests
- Fixed OSGi imports and exports

gokul commented:
Closing PR since this functionality is not needed just yet for DAS 4.
